Indicators/SqueezeMomentum.cs

```diff
@@ -0,0 +1,109 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using QuantConnect.Data.Market;
+
+namespace QuantConnect.Indicators
+{
+    /// <summary>
+    /// The SqueezeMomentum indicator calculates whether the market is in a "squeeze" condition,
+    /// determined by comparing Bollinger Bands to Keltner Channels. When the Bollinger Bands are
+    /// inside the Keltner Channels, the indicator returns 1 (squeeze on). Otherwise, it returns -1 (squeeze off).
+    /// </summary>
+    public class SqueezeMomentum : BarIndicator, IIndicatorWarmUpPeriodProvider
+    {
+        /// <summary>
+        /// The Bollinger Bands indicator used to calculate the upper, lower, and middle bands.
+        /// </summary>
+        private readonly BollingerBands _bollingerBands;
```

Comment:
That's what I initially did, but the implemented Keltner Channels use "typicalPrice = (input.High + input.Low + input.Close)/3" instead of just "closePrice." I looked for information on why typicalPrice would be used, but I couldn't find a clear reason. That's why I'm manually calculating the Keltner Channels.